backend/src/event/Event.ts

```diff
@@ -0,0 +1,314 @@
+import { EventProtocol } from '@entity/EventProtocol'
+import decimal from 'decimal.js-light'
+import { EventProtocolType } from './EventProtocolType'
+
+/*
+export interface EventInterface {
+  type: string
+  createdAt: Date
+  userId: number
+  xUserId?: number
+  xCommunityId?: number
+  transactionId?: number
+  contributionId?: number
+  amount?: Decimal
+}
+*/
+
+export class EventBasic {
```

Comment:
This is a lot of code. Why so complicated. What is the advantage of having this detailed definition? The result written to the DB is the same.

Reply:
yes that's right, but you get next to the typesaftyness with the correct EnumEventTypes a control which attributes of an event has to be or can be set for this EventType. Additionally the event classes are able to collect event attributes during the business logic before the event itself will be written. 
So these classes and detailed definitions ensures a higher quality of a written event and reduce the responsibility by the developer